flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -383,16 +427,43 @@ private boolean allSplitsFinished() {
 
     private void splitChunksForRemainingTables() {
         try {
+            // restore from a checkpoint
+            if (splittingTableId != null) {
+                chunkSplitter.switchSplittingTable(splittingTableId);
+                splitUnevenlySizedChunks(splittingTableId);
```

Comment:
Why restored chunkSplitter must be unEvenly here? The logic is a little confused.

Reply:
Only unEvenly table will use the state to store the information. Evenly table will split the table only once and get all of the splits.